bluetooth: services: ras: Add RREQ implementation to RAS

sean-madigan opened the pull request:
Add implementation for Ranging Requestor, a Ranging Service client which can be used to obtain channel sounding ranging data from a Ranging Responder which operates as a Ranging Service server.

This current implementation can perform most mandatory features from the Ranging Profile/Ranging Service
specifications to alert an application of available ranging data and receive the data from the server.

Further work can be done to add support for reading the features from the server, receiving real time ranging data, aborting receive operation, adding filters for ranging data.

Vge0rge committed:
nrf_security: CTR_DRBG needs to depend on random

Make the PSA_NEED symbol for CTR_DRBG with CRACEN
to depend on the PSA_WANT_GENERATE_RANDOM since this
is what it used for.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

Vge0rge committed:
nrf_security: Loosen the deps for builtin keys

Separate the builtin keys from the key management
driver since the builtin keys don't strictly require
the driver.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

Vge0rge committed:
nrf_security: Ifdef encrypted keys in KMU

This makes the KMU driver more configurable in the
sense that it allows the encryption for KMU keys
to be optionally enabled based on the enabled
configurations.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

Vge0rge committed:
nrf_security: Enable ED25519 with KMU

Enable the option
PSA_NEED_CRACEN_KEY_TYPE_ECC_TWISTED_EDWARDS_255 when
the KMU driver is enabled since KMU supports storing
keys of this type.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

Vge0rge committed:
nrf_security: Guard IKG function with key type

The IKG in Cracen only supports one asymmetric key
type, the SECP256R1. This guards the function call
to the IKG with the relevant configuration. This is
important because the PSA signing APIs can be used
with algorithms like EDDSA which have lower footprint
so it can reduce the flash usage when ECDSA is not
needed.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

michalek-no committed:
fw_info: Change allowed offsets for firmware info struct

Since the vector table of nRF54LX series is larger than before using it
is nice to have an offset between `0x400` and `0x800` to store `fw_info`
in to ensure you can tightly pack your application together with
firmware info. One downside of this is that this new offset is not
supported in any older version of NSIB.

Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>

michalek-no committed:
cmake: partition_manager: Add OTP region in partition manager for nRF54L

Added OTP region in partition manager for `nRF54L15X_ENGA` devices and
modified the secure boot storage partition to use OTP for NRF54L15.

Ref. NCSDK-25306

Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>

michalek-no committed:
bl_storage: fw_info/counters port to RRAMC

Changed the implementation of bl_storage to be compatible with RRAMC and
FLASH/NVMC. Due to the limitations in the RRAM controller OTP can only
be written to by words not half-words as it can be in NVMC.

This means we need to change the way we handle certain cases in
`bl_storage.c`, `bl_validation.c` and `fw_info.c`. This commit moves
some of the abstractions into two separate files `bl_storage_nvmc.c` and
`bl_storage_rramc.c`. However there are multiple other changes needed
which are done by using `#ifdef`'s.

Ref. NCSDK-25306

Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>

michalek-no committed:
cmake: provision: Change provision_hex to account for larger LCS

Changed the provisioning script to account for larger Life Cycle State
structures. Since we can only do word writes in nRF54LX life cycle
states needs to be stored in words. This increased the LCS struct by 1
word requiring us to change the way the provisioning script works.

This change adds the LCS size as a value to the CMake file which uses
the SOC to decide which size the LCS struct is which is then passed to
the python script through the `--lcs-state-size` argument.

Ref. NCSDK-25306

Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>
Signed-off-by: Sigurd Hellesvik <sigurd.hellesvik@nordicsemi.no>

michalek-no committed:
bootloader: Fix up lacking fprotect guards

Instead of just failing on compilation this commit changes fprotect
requirements so that it prints out a warning and error in log.

Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>

michalek-no committed:
Kconfigs: adjust selection of NRFX_NVMC

NRFX_NVMC is being selected conditionally as it is
not the one and only memory backend anymore.

Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>

umapraseeda committed:
doc: Doc fixes for 2.8

Doc fixes for 2.8 - release task

Signed-off-by: Uma Praseeda <uma.praseeda@nordicsemi.no>

NordicBuilder committed:
manifest: Update doc-internal revision (auto-manifest PR)

Automatically created by Github Action

Signed-off-by: Nordic Builder <pylon@nordicsemi.no>

morgendagen committed:
zigbee: fix error when using VERSION file in Zigbee project

Allow tweaks in APPLICATION_VERSION_STRING, e.g. 1.0.0+3.

Signed-off-by: Peder Toftegaard Olsen <peder@morgendagen.dk>
Co-authored-by: Wille Backman <wille.backman@nordicsemi.no>

annwoj committed:
doc: flpr and building for nrf54l15

Adds building article
Adds FLPR intro

Signed-off-by: Anna Wojdylo <anna.wojdylo@nordicsemi.no>

de-nordic committed:
sysbuild: MCUboot with ED25519 and KMU via PSA support

The commit will enforce building nrf54l15 with PSA enabled
ED25519, with CONFIG_NRF_SECURITY=y.
The commit adds SB_CONFIG_MCUBOOT_SIGNATURE_USING_KMU that allows
to build MCUboot for signature verification via KMU instead of
compiled in keys.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

michalek-no committed:
lib: fprotect: enable RRAMC low level backend

Selects NRFX_RRAMC symbol required by fprotect RRAM backend
on 54L15 platform.

Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>

michalek-no committed:
tests: bl_storage: nrf54l15

Adds bl_storage tests.

Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>

gmarull committed:
bluetooth: services: lbs: button_state only used if poll is enabled

The global button_state variable is only used in
CONFIG_BT_LBS_POLL_BUTTON=y.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

NordicBuilder committed:
manifest: Update sdk-zephyr revision (auto-manifest PR)

Automatically created by Github Action

Signed-off-by: Nordic Builder <pylon@nordicsemi.no>

NordicBuilder committed:
manifest: Update sdk-zephyr revision (auto-manifest PR)

Automatically created by Github Action

Signed-off-by: Nordic Builder <pylon@nordicsemi.no>

ahasztag committed:
suit: Fix not working SHA-512

This fixes the error causing manifests with SHA-512 installation
failure.

Signed-off-by: Artur Hadasz <artur.hadasz@nordicsemi.no>

thst-nordic committed:
quarantine: wifi provisioning tests on macOS

there is a problem with nanoPb that needs to be resolved

Signed-off-by: Thomas Stilwell <thomas.stilwell@nordicsemi.no>

richabp committed:
doc: wifi: Update memory footprint tables for 2.8 release

Updated Wi-Fi memory footprint tables for the 2.8.0 release.

SHEL-3194

Signed-off-by: Richa Pandey <richa.pandey@nordicsemi.no>

greg-fer committed:
doc: bootloader_partitioning: add info about VSC

Edited the structure of bootloader_partitioning page to mention
memory reports of the VSC extension alongside partition_manager_report.
Fixed some formatting and structure issues and typos.
VSC-2712.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

plskeggs committed:
doc: Update code owners to remove incorrect assignments

@plskeggs was incorrectly added as a code owner for
/modules/azure-sdk-for-c and for/modules/memfault-firmware-sdk.

Remove those, and change the remaining assignment on
/modules/cjson to @nrfconnect/ncs-nrf-cloud.

Signed-off-by: Pete Skeggs <peter.skeggs@nordicsemi.no>

divipillai committed:
doc: Add info about ZMS in the migration guide

Add info about ZMS in the migration guide.

Signed-off-by: divya pillai <divya.pillai@nordicsemi.no>

nordic-krch committed:
samples: peripheral: lpuart: Remove redundant entries in overlays

Remove enabling of rtc node. RTC node is not used in the sample.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
samples: peripheral: lpuart: Move configuration to a board conf

Clean prj.conf and move instance specific configuration to a board
specific conf file.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
drivers: serial: nrf_sw_lpuart: Add support for nrf54x

Do not use NRF_GPIOTE register as new platforms may have multiple
GPIOTE instances. Add support to a case when there are GPIO ports
which cannot work with GPIOTE. Pins from that ports cannot be used
for control pins in low power UART.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
samples: peripheral: lpuart: Add support for nrf54l15dk

Add configuration for nrf54l15dk//cpuapp target.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
samples: peripheral: lpuart: Cleanup in debug configuration

Simplify debug configuration. Remove debug conf and overlay file.
All targets are now configuring console UART to be in the lowest
power state when not transmitting logs. With this configuration it
is possible to have console uart enabled in debug and none debug
build. It is also possible to reach the lowest energy state in
the debug build, too.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nvlsianpu committed:
manifest: zephyr update

added nRF54l15dk fir zephyr's smp_svr's sample.yaml

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

nvlsianpu committed:
doc/dev_guides/nrf54l: added fota_update guide

Added fota_update guide for nRF54l15 SoC based on the nRF52 doc.

ref.: NCSDK-29601

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>
Signed-off-by: Anna Wojdylo <anna.wojdylo@nordicsemi.no>
Signed-off-by: Omkar Kulkarni <omkar.kulkarni@nordicsemi.no>

nordic-krch committed:
drivers: serial: nrf_sw_lpuart: Remove RX timeout

Low power UART implements asynchronous UART API and uart_rx_enable
has timeout as one of the parameters. However, low power UART has
packets delimited by the control GPIO pin toggling so timeout here
does not make much sense and it is not expected that user will get
data when UART RX timeout occur (it is expected that data is
received when transmitter ends the packet by deasserting control pin.

Given that, timeout provided to the low power UART uart_rx_enable
call must not be used when underneath UART device is controlled.
Low power UART must open receiver without any timeout as reception
is terminated by the control pin deassertion.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
tests: drivers: lpuart: Cleanup in conf files

Move common part from board specific conf file to the test conf
file.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
tests: drivers: lpuart: Add nrf54l15dk//cpuapp

Add nrf54l15dk/nrf54l15/cpuapp configuration to the test.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

tomchy committed:
tests: Add clang-format switches

Add clang-format switches in SUIT storage tests.

Ref: NCSDK-29626

Signed-off-by: Tomasz Chyrowicz <tomasz.chyrowicz@nordicsemi.no>

tomchy committed:
tests: Add SUIT storage purge tests

Add tests that verify per-domain SUIT storage purge API.

Ref: NCSDK-29626

Signed-off-by: Tomasz Chyrowicz <tomasz.chyrowicz@nordicsemi.no>

nordic-segl committed:
tests: drivers: pwm: gpio_loopback: Enable test on PWM120 instance

Check if fast PWM instance PWM120 is working correctly.
Check that PWM signal can be generated on all four PWM channels.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

greg-fer committed:
doc: nrfvsc: edits to general docs

Edited sections about the nRF Connect for VSC in the general docs.
Edited information about which apps sysbuild is enabled by default for.
VSC-2711.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Add HID reportq documentation

Change introduces documentation for the HID report queue utility.

Jira: NCSDK-28433

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>
Signed-off-by: Pekka Niskanen <pekka.niskanen@nordicsemi.no>

nordic-segl committed:
tests: benchmarks: multicore: idle_with_pwm: Enable test on PWM120

Confirm that target enters low power modes when PWM120
(fast PWM instance) is used.

Note:
Test doesn't confirm that PWM120 works correctly after leaving
low power state. It only measures supply current.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

umapraseeda committed:
doc: update tag file for 2.8

Update Zoomin tag file for 2.8

Signed-off-by: Uma Praseeda <uma.praseeda@nordicsemi.no>

krish2718 committed:
doc: nrf: Add a section about hostap

Now that we use upstream Zephyr fork for hostap, we need to document the
scope of features supported in NCS as it's only a subset of upstream
Zephyr.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

NordicBuilder committed:
manifest: Update sdk-zephyr revision (auto-manifest PR)

Automatically created by Github Action

Signed-off-by: Nordic Builder <pylon@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Update HID forward documentation

Change documents integrating `hid_report_subscriber_event` and HID
report queue utility. Commit contains also some small documentation
alignments.

Jira: NCSDK-28433

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>
Signed-off-by: Pekka Niskanen <pekka.niskanen@nordicsemi.no>

NordicBuilder committed:
manifest: Update sdk-zephyr revision (auto-manifest PR)

Automatically created by Github Action

Signed-off-by: Nordic Builder <pylon@nordicsemi.no>

mia-ko committed:
doc: fix maturity tables for crypto

Updates to Crypto API maturity table

Signed-off-by: Mia Koen <mia.koen@nordicsemi.no>

FrancescoSer committed:
doc: Updated 54h gs for 2.8

Updated 54h gs for 2.8

Signed-off-by: Francesco Domenico Servidio <francesco.servidio@nordicsemi.no>

adsz-nordic committed:
doc: nrf54h: Add SDFW update description

Add description of SUIT based update of:
- Secure Domain Firmware
- Secure Domain Firmware Recovery

Jira: NCSDK-27725

Signed-off-by: Adam Szczygieł <adam.szczygiel@nordicsemi.no>

kapi-no committed:
doc: software_maturity: fast_pair: update status for nRF54L15

Updated the software maturity levels in the Google Fast Pair category
for the nRF54L15 device to declare the "Supported" level for the locator
tag use case and its feature depedencies.

Ref: NCSDK-29887

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

FrancescoSer committed:
doc: 54H migration guide update

Updated migration guide for 2.8 with 54H info

Signed-off-by: Francesco Domenico Servidio <francesco.servidio@nordicsemi.no>

gmarull committed:
scripts: requirements: bump nrf-regtool to 8.0.0

It is required due to Zephyr changes.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

trantanen committed:
lib: sms: No warning log for unregistration failure

Change warning to debug log when unregistration (AT+CNMI=0,0,0,0) fails
as SMS might have been unregistered due to AT+CFUN=0,
or through AT+CNMI=0,0,0,0.

Jira: NRF91-2238

Signed-off-by: Tommi Rantanen <tommi.rantanen@nordicsemi.no>

NordicBuilder committed:
manifest: Update sdk-zephyr revision (auto-manifest PR)

Automatically created by Github Action

Signed-off-by: Nordic Builder <pylon@nordicsemi.no>

ArekBalysNordic committed:
samples: matter: Enable storing DAC priv key in KMU by default.

Enabled storing the DAC private key in KMU if it is available.

Signed-off-by: Arkadiusz Balys <arkadiusz.balys@nordicsemi.no>
Signed-off-by: Adrian Gielniewski <adrian.gielniewski@nordicsemi.no>

gchwier committed:
tests: boot_chains: enable nrf54l15dk

Add nrf54l15dk to boot_chains configurations.

Signed-off-by: Grzegorz Chwierut <grzegorz.chwierut@nordicsemi.no>

hellesvik-nordic committed:
partition_manager: Make tfm storage partions avoid partitioning conflict

With the previous method, both tfm_storage partitions (its, ps, etc)
were placed "before: mcuboot_primary". This could lead to a case
where mcuboot was placed after the tfm_storage partitions.
To avoid this, instead place "after" mcuboot or tfm_secure.

Jira: NCSDK-30094

Signed-off-by: Sigurd Hellesvik <sigurd.hellesvik@nordicsemi.no>

SylwesterKonczyk committed:
suit: SDFW update over sdfw_mirror

Copy directive checks a location of SDFW/SDFW_Recovery update
candidate and tries to install it using sdfw_mirror, if necessary

Signed-off-by: Sylwester Konczyk <sylwester.konczyk@nordicsemi.no>

nvlsianpu committed:
scripts/requirents-extra: add nrfprovision

Added requirement of nrfprovision script which is temporary
script needed for west ncs-provision script
(nRF54L15 KMU provisioning)

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

trantanen committed:
samples: cellular: nrf_cloud_multi_service: More modem RX shared memory

The sample reads certificates while checking their validity
to give better error messages.
PR #17189 optimized modem shared memory to 4kB for both TX and RX.
This is too small for RX with several certificates.
Issues seen with three Root CAs causing modem to eventually crash
as there is not enough RPC memory for the reponse.
So increasing CONFIG_NRF_MODEM_LIB_SHMEM_RX_SIZE by letting it use
default value of ~8kB.

Jira: TNSW-64434

Signed-off-by: Tommi Rantanen <tommi.rantanen@nordicsemi.no>

tomi-font committed:
manifest: tf-m: pull in RRAM improvements

Improvements to the duration and interrupt latency when writing
to RRAM.

Signed-off-by: Tomi Fontanilles <tomi.fontanilles@nordicsemi.no>

martintv committed:
mpsl: calibrate hfpll on 54L

On 54L hfpll can be calibrated, call mpsl callback to calibrate
even if LFXO is used and LFRC calibration is not needed.

Signed-off-by: Martin Tverdal <martin.tverdal@nordicsemi.no>

alxelax committed:
net: openthread: rpc: using nrf_rpc encoders and decoders

Commit adds refactoring openthread servers to use
nrf_rpc encoders and decoders instead of zcbor API.

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>

tokangas committed:
samples: cellular: gnss: Update GNSS factory almanacs

Updated GNSS factory almanacs.

Signed-off-by: Tommi Kangas <tommi.kangas@nordicsemi.no>

Vge0rge committed:
nrf_security: Mbedlts threading dependency fix

Make the MbedTLS threading configuration to depend on
the multithreading Kconfig option. The multithreading
configuration is required by both the PSA crypto
implementation which uses the Zephyr mutexes and
also by the MbedTLS legacy implementation which
may not use Zephyr mutexes for CC3XX.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

Damian-Nordic committed:
net: openthread: rpc: unit tests for ot_rpc_diag.c...

...and ot_rpc_netdata.c

1. Implement missing unit tests for OT API serialization
   implemented in ot_rpc_diag.c and ot_rpc_netdata.c
2. Fix discovered bugs.

Signed-off-by: Damian Krolik <damian.krolik@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Update HID forward documentation

Change documents CONFIG_DESKTOP_HID_FORWARD_SUBSCRIBER_COUNT Kconfig
option.

Jira: NCSDK-30002

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>
Signed-off-by: Pekka Niskanen <pekka.niskanen@nordicsemi.no>

michalek-no committed:
bootloader: bl_boot: uninit of UARTE20

adds another UART's uninit call.

Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>

alexstanoev-nordic committed:
bluetooth: services: ras: Add initial folder structure for ranging service

This adds the base folder structure and headers to be used in the
implementation of the Ranging Service draft specification.

Create RAS primary service and support reading RRSP features.

Signed-off-by: Aleksandar Stanoev <aleksandar.stanoev@nordicsemi.no>

alexstanoev-nordic committed:
CODEOWNERS: Make ncs-dragoon codeowner for ranging service

ncs-dragoon is the only codeowner for RAS during development.

Signed-off-by: Aleksandar Stanoev <aleksandar.stanoev@nordicsemi.no>

NordicBuilder committed:
manifest: Update sdk-nrfxlib revision (auto-manifest PR)

Automatically created by Github Action

Signed-off-by: Nordic Builder <pylon@nordicsemi.no>

PavelVPV committed:
samples: bluetooth: mesh: dfu: distributor: Disable RTT in mcuboot

RTT is used for interacting with the distributor over shell. Having it
enabled in mcuboot will require a user to manually provide correct
_SEGGER_RTT address otherwise RTT Viewer will pick wrong buffer and
shell will not work.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

PavelVPV committed:
samples: bluetooth: mesh: dfu: distributor: Enable RTT explicitly

This option is not enabled by default and apparently not enabled through
some other Kconfig options on nRF54L15. Enabling it explicitly to avoid
misconfiguration of the sample.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

FrancescoSer committed:
doc: nRF54H SUIT img update

Updated 54H SUIT doc images.

Signed-off-by: Francesco Domenico Servidio <francesco.servidio@nordicsemi.no>

nordic-bami committed:
tests: benchmarks: Start testing Fast UART with s2ram

This test uses P7 for UARTE120 instance

Signed-off-by: Bartosz Miller <bartosz.miller@nordicsemi.no>

maje-emb committed:
samples: peripheral_power_profiling: Fix the use of PM_DEVICE

Fixed the use of Device Power Management and
enabled Runtime Device Power Management.

Ref: NCSDK-30068

Signed-off-by: Marcin Jelinski <marcin.jelinski@nordicsemi.no>

ankuns committed:
samples: 802154_phy_test: remote_shell only for multicore

The APPCORE_REMOTE_SHELL option is selected by default only for
multicore devices. It is pointless to select this option for single
core devices.

Signed-off-by: Andrzej Kuros <andrzej.kuros@nordicsemi.no>

divipillai committed:
doc: Add sdk nrf toolchain docker image

Added nRF connect SDK docker image.

Signed-off-by: divya pillai <divya.pillai@nordicsemi.no>

adsz-nordic committed:
suit: Rework SDFW and SDFW Recovery sinks

New implementation of SDFW and SDFW Recovery sinks.
Supports fixed installation order (SDFW, then SDFW Recovery).
Treats SDFW Recovery installation as optional.
Envelope is saved when SDFW slot is successfully installed.

Ref: NCSDK-29561

Signed-off-by: Adam Szczygieł <adam.szczygiel@nordicsemi.no>

alxelax committed:
net: openthread: rpc: using common decoding error handler

Using common decoding error handler for all commands
and responses.

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>

alxelax committed:
net: openthread: rpc: revert get_uint_t decoder

Commit reverts get_uint_t decoder to keep client
endianness tolerant.

Signed-off-by: Aleksandr Khromykh <aleksandr.khromykh@nordicsemi.no>

maciejbaczmanski committed:
doc: openthread: update otbr version

update to newest otbr available in dockerhub

Signed-off-by: Maciej Baczmanski <maciej.baczmanski@nordicsemi.no>

maciejbaczmanski committed:
doc: matter: update matter version and spec table

update matter version and spec table for release

Signed-off-by: Maciej Baczmanski <maciej.baczmanski@nordicsemi.no>

nvlsianpu committed:
sysbuild/mcuboot: ed25519 pure signature integration

Added integration of signing image with the pure signature:
- New SB_CONFIG_BOOT_SIGNATURE_TYPE_PURE switch for enabling pure
signature.
- enforced sha512 for ed25519 on nRF54l target
- requests pure signature from imgtool.py

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>
Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

masz-nordic committed:
test-spec: align to namechange

`emulated` has been changed to `sdp`.

Signed-off-by: Marcin Szymczyk <marcin.szymczyk@nordicsemi.no>

gmarull committed:
tests: replace option to disable MRAM latency control

This is now handled by a SoC level "service".

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

jangalda-nsc committed:
requirements: Use Nordic pip-compile-cross-platform

Nordic version of pip-compile-cross-platform adds support for
propagating extra-index-urls entries to requirements-fixed.txt

Signed-off-by: Jan Gałda <jan.galda@nordicsemi.no>
Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

rado17 committed:
net: lib: wifi_credentials: Change argument type for MFP

Modify the arguments expected by MFP configuration in
`wifi_cred add` command.
Fixes SHEL-3216.

Signed-off-by: Ravi Dondaputi <ravi.dondaputi@nordicsemi.no>

rugeGerritsen committed:
samples: Bluetooth: Initialize conn to NULL for bt_conn_le_create()

After https://github.com/zephyrproject-rtos/zephyr/pull/79856
a warning message is printed out if the connection object is not
initialized to NULL.

This commit addresses this by initialing it.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

rugeGerritsen committed:
bluetooth: Initialize conn to NULL for bt_conn_le_create()

After https://github.com/zephyrproject-rtos/zephyr/pull/79856
a warning message is printed out if the connection object is not
initialized to NULL.

This commit addresses this by initialing it.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

nordicjm committed:
manifest: Update sdk-mcuboot

Pulls in a fix for decompression

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

rugeGerritsen committed:
tests: bluetooth: Initialize conn to NULL for bt_conn_le_create()

After https://github.com/zephyrproject-rtos/zephyr/pull/79856
a warning message is printed out if the connection object is not
initialized to NULL.

This commit addresses this by initialing it.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

rugeGerritsen committed:
applications: nrf5340_audio: Set conn to NULL for bt_conn_le_create()

After https://github.com/zephyrproject-rtos/zephyr/pull/79856
a warning message is printed out if the connection object is not
initialized to NULL.

This commit addresses this by initialing it.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

MarekPieta committed:
scripts: nrf_profiler: Fix usage of deprecated `np.float`

The `np.float` was removed (it was a deprecated alias for `float`).

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

NordicBuilder committed:
manifest: Update sdk-nrfxlib revision (auto-manifest PR)

Automatically created by Github Action

Signed-off-by: Nordic Builder <pylon@nordicsemi.no>

ankuns committed:
samples: 802154_phy_test: fix for nRF54H20_ENGB

The app_rpc.c code need to be aware of CONFIG_SOC_NRF54H20_ENGB_CPURAD.

Signed-off-by: Andrzej Kuros <andrzej.kuros@nordicsemi.no>

maje-emb committed:
samples: peripheral_power_profiling: Remove unused buttons from DTS

Removed unused buttons from DTS, no longer wake up the application.
This ensures that these buttons are not initialized by the
dk_buttons_and_leds library.

Ref: NCSDK-30039

Signed-off-by: Marcin Jelinski <marcin.jelinski@nordicsemi.no>

maciejbaczmanski committed:
doc: openthread: update power measurement tables

update power measurement tables and add nrf54l15 to them

Signed-off-by: Maciej Baczmanski <maciej.baczmanski@nordicsemi.no>

nordicjm committed:
doc: nrf: app_dev: bootloaders_dfu: Fix compression static PM files

Links to files in the sample rather than the test

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordic-segl committed:
tests: benchmarks: multicore: idle_with_pwm: Disable GD2 and GD3 manually

This commit shall be dropped when GD handling
is implemented in sdk-zephyr.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

jangalda-nsc committed:
oss-history: Upgrade pip in GH action

Fixes issue with pynrfjprog installation

Signed-off-by: Jan Gałda <jan.galda@nordicsemi.no>
Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

nordic-segl committed:
tests: benchmarks: multicore: idle_spim_loopback: Add test for SPIM120

Add test that combinest functional check (whether SPIM transmits data)
with non-functional check (current consumption).

Confirm that slow SPIM130 / fast SPIM120:
- doesn't block entering low power states like IDLE or S2RAM;
- work correclty after leaving low power state.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

nordic-segl committed:
tests: benchmarks: multicore: idle_spim_loopback: Disable GD2 and GD3

This commit shall be dropped when GD handling
is implemented in sdk-zephyr.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

simensrostad committed:
samples: matter: light_bulb: Fix AWS IoT configuration

Enable SHA384 to fix connection towards AWS IoT.
SHA384 isn't required by AWS IoT, but due to some configuration issues
in MBEDTLS, its needed.

Signed-off-by: Simen S. Røstad <simen.rostad@nordicsemi.no>

NordicBuilder committed:
manifest: Update sdk-find-my revision (auto-manifest PR)

Automatically created by Github Action

Signed-off-by: Nordic Builder <pylon@nordicsemi.no>

maciejbaczmanski committed:
samples: matter: fix fprotect for mcuboot

Disable `FPROTECT` in app and reenable it in mcuboot for nrf54l15

Signed-off-by: Maciej Baczmanski <maciej.baczmanski@nordicsemi.no>

NordicBuilder committed:
manifest: Update sdk-zephyr revision (auto-manifest PR)

Automatically created by Github Action

Signed-off-by: Nordic Builder <pylon@nordicsemi.no>

nvlsianpu committed:
.github/workflows/west-commands: upgrade pip

Added step for upgrade pip before installing py requirements.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

gmarull committed:
samples: suit: flash_companion: disable GPD service

Otherwise flash overflows.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

frkv committed:
doc: crypto: Correct thread-safety for PSA crypto APIs

-Mbed TLS 3.6.0 adds thread safety for PSA crypto APIs, but this has
 not been reflected in the documentation. Removed a caution related
 to this.

Signed-off-by: Frank Audun Kvamtrø <frank.kvamtro@nordicsemi.no>

frkv committed:
doc: crypto: Removing reference to use custom configs for Mbed TLS

-We don't support providing a pre-written Mbed TLS config file
 while building nrf_security. The Kconfig's used for configurations
 are parsed and handled in CMake logic and there is no standard way
 to extract information from a pre-written header-file and have that
 lead to changed CMake logic.

Signed-off-by: Frank Audun Kvamtrø <frank.kvamtro@nordicsemi.no>

FrancescoSer committed:
doc: 54H J-Trace PRO compatibility

Updated 54H logging guide.

Signed-off-by: Francesco Domenico Servidio <francesco.servidio@nordicsemi.no>

frkv committed:
doc: crypto: Document deprecation of legacy Mbed TLS APIs

-This commit adds cautions that legacy Mbed TLS crypto toolbox APIs
 are deprecated and will be removed.
-This also simplifies a statement around enabling both nrf_security
 and nordic_security_backend. This is not required as
 nrf_security will be automatically set once nordic_security_backend
 is set (which is soon to be deprecated and removed)

ref: NCSDK-30036

Signed-off-by: Frank Audun Kvamtrø <frank.kvamtro@nordicsemi.no>

maciejbaczmanski committed:
doc: update software maturity levels for ot and matter

update software maturity levels for ot and matter

Signed-off-by: Maciej Baczmanski <maciej.baczmanski@nordicsemi.no>

maciejbaczmanski committed:
samples: matter: fix wrong cluster revision for On/Off

"On/Off" clusters should have revision 6

Signed-off-by: Maciej Baczmanski <maciej.baczmanski@nordicsemi.no>

nika-nordic committed:
samples: peripheral_power_profiling: nrf54l: disable RAM retention

Disabled RAM retention saves energy in System OFF mode
and can be used to showcase nRF54L15 low power consumption.

Signed-off-by: Nikodem Kastelik <nikodem.kastelik@nordicsemi.no>

RobertGalatNordic committed:
manifest: update sidewalk revision

update sidewalk revision

Signed-off-by: Robert Gałat <robert.galat@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Prevent enabling HF clock lock on nRF54H

Add a Kconfig dependency to prevent enabling HF clock lock on nRF54H SoC
Series. The hotfix does not support this SoC Series. The commit also
improves Kconfig help of the option.

Jira: NCSDK-29545

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

michalek-no committed:
bootloader: bl_boot: minor ifdef rework

Changing ifdef's for consistency improvement.
This also spares few bytes of flash.

Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>

adigie committed:
doc: matter: Update default storage for DAC key on nRF54L15

Change default storage for DAC key on nRF54L15 to KMU.

Signed-off-by: Adrian Gielniewski <adrian.gielniewski@nordicsemi.no>

Commit:
doc: Mesh maturity to supported for nRF54L15

Changed the software maturity level for Bluetooth Mesh
to Supported for the nRF54L15.

Signed-off-by: Ingar Kulbrandstad <ingar.kulbrandstad@nordicsemi.no>

annwoj committed:
doc: update mcuboot support info

Updated mcuboot section in features of.
Added link to the nrf54L15 datasheet.

Signed-off-by: Anna Wojdylo <anna.wojdylo@nordicsemi.no>

sean-madigan committed:
bluetooth: services: ras: Add RREQ implementation to RAS

Add implementation for Ranging Requestor, a Ranging Service
client which can be used to obtain channel sounding ranging
data from a Ranging Responder which operates as a Ranging
Service server.

This current implementation can perform most mandatory
features from the Ranging Profile/Ranging Service
specifications to alert an application of available ranging
data and receive the data from the server.

Further work can be done to add support for reading the
features from the server, receiving real time ranging data,
aborting receive operation, adding filters for ranging data

Signed-off-by: Sean Madigan <sean.madigan@nordicsemi.no>

alexstanoev-nordic commented on include/bluetooth/services/ras.h:
```diff
@@ -130,6 +131,61 @@ struct ras_subevent_header {
 } __packed;
 BUILD_ASSERT(sizeof(struct ras_subevent_header) == BT_RAS_SUBEVENT_HEADER_LEN);
 
+typedef void (*bt_ras_rreq_rd_ready_cb_t)(struct bt_conn *conn, uint16_t ranging_counter);
```
Will be nice to have some docs for these

alexstanoev-nordic commented on include/bluetooth/services/ras.h:
```diff
@@ -130,6 +131,61 @@ struct ras_subevent_header {
+
+/** @brief Get ranging data for given ranging counter.
+ *
+ * @note This should only be called after receiving a ranging data ready notification/indication and
```
Maybe this is more like what it'll look like to the app?
```suggestion
 * @note This should only be called after receiving a ranging data ready callback and
```

alexstanoev-nordic commented on subsys/bluetooth/services/ras/rreq/Kconfig.ras_rreq:
```diff
@@ -17,4 +17,11 @@ module = BT_RAS_RREQ
 module-str = RAS_RREQ
 source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config"
 
+config BT_RAS_MAX_ACTIVE_RREQ
```
This will match what I've done for RRSP
```suggestion
config BT_RAS_RREQ_MAX_ACTIVE_CONN
```

alexstanoev-nordic commented on subsys/bluetooth/services/ras/rreq/ras_rreq.c:
```diff
@@ -0,0 +1,723 @@
+	int err;
+
+	struct ras_ack_ranging_data ack_ranging_data;
+	ack_ranging_data.opcode = RASCP_OPCODE_ACK_RD;
```
You could also avoid having these structs by doing something like
```
	NET_BUF_SIMPLE_DEFINE(rsp, RASCP_CMD_OPCODE_LEN + sizeof(uint16_t));

	net_buf_simple_add_u8(&rsp, RASCP_OPCODE_ACK_RD);
	net_buf_simple_add_le16(&rsp, rreq->on_demand_rd.counter_in_progress);
```

alexstanoev-nordic commented on subsys/bluetooth/services/ras/rreq/ras_rreq.c:
```diff
@@ -0,0 +1,723 @@
+	uint8_t segmentation_header = net_buf_simple_pull_u8(&segment);
+
+	bool first_segment = segmentation_header & BIT(0);
+	bool last_segment = segmentation_header & BIT(1);
```
Might be nice to explicitly make this a bool
```suggestion
	bool last_segment = !!(segmentation_header & BIT(1));
```

olivier-le-sage replied:
isn't it safe ? assigning/casting to bool

alexstanoev-nordic replied:
Yup apparently with stdbool and _Bool this is fine, only matters for pre-c99 so disregard my comment

alexstanoev-nordic commented on subsys/bluetooth/services/ras/rreq/ras_rreq.c:
```diff
@@ -0,0 +1,723 @@
+		if (err != 0 && err != -EINVAL) {
+			LOG_DBG("Failed to unsubscribe to RAS-CP: %d", err);
+
+			return err;
```
Any of these failing will leave a ref to the conn, maybe the errors should be ignored as the link is probably getting torn down?

sean-madigan replied:
Good point

nordic-pikr committed:
tests: benchmarks: peripheral_load: add clock control

Add clock change requests to peripheral_load test

Signed-off-by: Piotr Krzyzanowski <piotr.krzyzanowski@nordicsemi.no>

krish2718 committed:
doc: nrf70: Fix 54 Series shields

The shields usage is now updated.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

gmarull committed:
samples: suit: flash_companion: disable MRAM latency service

Otherwise NRFS can't be disabled.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

Damian-Nordic committed:
net: openthread: rpc: don't bring interface up in IF_ENABLE command

The OpenThread RPC server starts the network interface
automatically on boot because otherwise no frames are
delivered to the OpenThread stack.

Therefore, don't bring the interface up or down when
IF_ENABLE RPC command is received. Instead, assume that
IF_ENABLE command only indicates that the client has
brought its interface up or down, and that it is
interested (or not) in receiving all incoming packets.

Signed-off-by: Damian Krolik <damian.krolik@nordicsemi.no>

greg-fer committed:
doc: zigbee: NCP Host v2.2.4

Updated NCP Host version shortcut to v2.2.4.
Updated ZBOSS version shortcut to 3.11.5.0.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

jhirsi committed:
samples: dect_phy: dect_shell: new startup_cmd added

New command to store shell commands
to be run sequentially after a bootup.
Default max of 6 mem slots available for commands.

Signed-off-by: Jani Hirsimäki <jani.hirsimaki@nordicsemi.no>

jhirsi committed:
samples: dect: dect_phy: dect_shell: enabled CONFIG_AT_SHELL

Enabled to give at commands from shell prompt.
Note that NR+ modem FW has rather limited set of AT commands supported.

Signed-off-by: Jani Hirsimäki <jani.hirsimaki@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Depracate HF clock lock

Mark the HF clock lock hotfix as deprecated. Setting the peripheral
latency LLPM connection parameter to `0` leads to keeping the HF clock
enabled.

Jira: NCSDK-29545

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

nordic-bami committed:
tests: benchmarks: Add test for s2ram with uart automatic usage

Test UARTE driver power management without manual enable/disable

Signed-off-by: Bartosz Miller <bartosz.miller@nordicsemi.no>

D-Triveni committed:
samples: wifi: Optimize network buffer settings for improved throughput

Override default 4KB sizes of Tx and Rx data pools to achieve
better throughput.

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

bugo-nordic commented on subsys/bluetooth/services/ras/rreq/ras_rreq.c:
```diff
@@ -0,0 +1,707 @@
+	uint8_t segmentation_header = net_buf_simple_pull_u8(&segment);
+
+	bool first_segment = segmentation_header & BIT(0);
+	bool last_segment = !!(segmentation_header & BIT(1)) ;
```
What's the reason for `!!`? Does casting to (bool) not help?

maje-emb committed:
samples: peripheral_power_profiling: Add Kconfig option to disable LEDs

Add Kconfig option to disable LEDs for power saving.

Signed-off-by: Marcin Jelinski <marcin.jelinski@nordicsemi.no>

edmont committed:
doc: zigbee: update memory requirement tables

Update Zigbee samples memory requirements and stack usage.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

NordicBuilder committed:
manifest: Update sdk-find-my revision (auto-manifest PR)

Automatically created by Github Action

Signed-off-by: Nordic Builder <pylon@nordicsemi.no>

mkapala-nordic committed:
samples: bluetooth: fast_pair: locator_tag: Drop MBEDTLS_CFG_FILE

The CONFIG_MBEDTLS_CFG_FILE Kconfig is enabled by default in the
MCUboot image and does not require explicit configuration. The explicit
configuration also throws an error for non-default MCUboot signatures
(such as RSA) that do not depend on the MBEDTLS Kconfig and do not
enable it.

Jira: NCSDK-29943

Signed-off-by: Mateusz Kapala <mateusz.kapala@nordicsemi.no>

nordic-piks committed:
tests: benchmarks: power_consumption: qdec: enable PM_DEVICE

Define sleep configuration.

Signed-off-by: Piotr Kosycarz <piotr.kosycarz@nordicsemi.no>

krish2718 committed:
doc: nrf: Fix 54L PDK to DK

54L PDK support is now removed.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

hakonfam committed:
doc: nrf54h20: add description of sdfw bootstatus

Ref: NCSDK-30107
Signed-off-by: Håkon Amundsen <haakon.amundsen@nordicsemi.no>

trantanen committed:
samples: cellular: Change deprecated OVERLAY_CONFIG and DTC_OVERLAY_FILE

Replace deprecated OVERLAY_CONFIG with EXTRA_CONF_FILE.
Replace deprecated DTC_OVERLAY_FILE with EXTRA_DTC_OVERLAY_FILE.

Signed-off-by: Tommi Rantanen <tommi.rantanen@nordicsemi.no>

nordic-piks committed:
tests: benchmarks: current_consumption: system_off: adjust test fun

Different thresholds are needed.

Signed-off-by: Piotr Kosycarz <piotr.kosycarz@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Add LLPM dependency to HF clock lock

Change adds LLPM dependency to HF clock lock hotfix. Using LLPM
connection parameters reduces HID data latency far more significantly
than enabling the hotfix.

Jira: NCSDK-29545

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

tomi-font committed:
manifest: tf-m: RRAM fixup

-

Signed-off-by: Tomi Fontanilles <tomi.fontanilles@nordicsemi.no>

nordic-bami committed:
tests: benchmarks: increase idle HPU feature test timeout

Default 60s is to low and the test timed out, set to 90s

Signed-off-by: Bartosz Miller <bartosz.miller@nordicsemi.no>

mkapala-nordic committed:
samples: bluetooth: fast_pair: locator_tag: Set RSA sig type for DKs

Set signature type to RSA for DKs.

Jira: NCSDK-29943

Signed-off-by: Mateusz Kapala <mateusz.kapala@nordicsemi.no>

jnsgsr committed:
doc: samples: SUIT: Update SMP Transfer Sample guide

Update after the manual tests for the release.

Signed-off-by: Janusz Gąsior <janusz.gasior@nordicsemi.no>

krish2718 committed:
doc: sml: Mark promiscuous mode as supported

This is now a supported feature.

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

MarekPieta committed:
caf: ble_state: Block enabling CONFIG_CAF_BLE_USE_LLPM on nRF53

Low Latency Packet Mode is not supported on the nRF53 SoC Series.

Jira: NCSDK-29545

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>
Signed-off-by: Divya Pillai <divya.pillai@nordicsemi.no>

LuDuda committed:
applications: matter_bridge: Increase heap size

Increase heap size to accommodate the memory required for additional
bridged devices.

Signed-off-by: Łukasz Duda <lukasz.duda@nordicsemi.no>

maciejbaczmanski committed:
samples: matter: build nrf54h20 without migration manager

build nrf54h20 without migration manager as it is not ready
for this platform yet.

Signed-off-by: Maciej Baczmanski <maciej.baczmanski@nordicsemi.no>

trantanen committed:
samples: cellular: location|mosh|atv2: Modem UART traces with Wi-Fi

CONFIG_NRF_MODEM_LIB_TRACE=n had been defined in nrf700x overlay
because that was needed long time ago to make nrf91+nrf7002ek to work.
Removing it now to allow use of modem UART traces with Wi-Fi builds
utilizing nrf91-modem-trace-uart snippet.

Signed-off-by: Tommi Rantanen <tommi.rantanen@nordicsemi.no>

jtguggedal committed:
applications: connectivity_bridge: Increase buffers for Thingy:91 X

Increase buffer count for Thingy:91 X to be able to capture
modem traces without running out of memory.

Signed-off-by: Jan Tore Guggedal <jantore.guggedal@nordicsemi.no>

de-nordic committed:
tests: kmu: Switch from MCUboot Kconfigs to sysbuild

Select KMU via sysbuild.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

NordicBuilder committed:
manifest: Update sdk-nrfxlib revision (auto-manifest PR)

Automatically created by Github Action

Signed-off-by: Nordic Builder <pylon@nordicsemi.no>

maciejbaczmanski committed:
doc: matter: fix memory requirements for 54l15 TF-M

Fix memory requirements for 54l15 TF-M to take into account
RAM and ROM usage by TF-M itself

Signed-off-by: Maciej Baczmanski <maciej.baczmanski@nordicsemi.no>

kszromek-nordic committed:
manifest: update suit-generator revision

suit-generator with default ImageCreator adresses for nRF54H20 engC

Ref: NCSDK-30031

Signed-off-by: Krzysztof Szromek <krzysztof.szromek@nordicsemi.no>

FrancescoSer committed:
doc: Updated 54H samples build commands

Updated 54H samples build commands for Eng B.

Signed-off-by: Francesco Domenico Servidio <francesco.servidio@nordicsemi.no>

MarekPieta committed:
applications: nrf_desktop: Update HF clock lock documentation

Change updates HF clock lock module documentation. Change adds
information about lack of nRF54H SoC Series support, LLPM dependency
and module deprecation.

Jira: NCSDK-29545

Signed-off-by: Marek Pieta <Marek.Pieta@nordicsemi.no>

alexstanoev-nordic committed:
bluetooth: services: ras: Implement Ranging Responder role of RAS

Implement the ranging responder (RRSP) server role of the
Ranging Service draft specification.
This allows a ranging requestor (RREQ) to retrieve channel sounding
ranging data from the local device.

This implementation contains support for all mandatory features,
as well as using notifications for data transfer.

Signed-off-by: Aleksandar Stanoev <aleksandar.stanoev@nordicsemi.no>

karun2796 committed:
manifest: Update sdk-nrfxlib

Update firmware blobs to fix EAP AKM.

Signed-off-by: Karun Kumar Eagalapati <karun.kumar@nordicsemi.no>

nordic-krch committed:
samples: peripheral: lpuart: Fix uart0 pins in nrf52840dk and nrf21540dk

Wrong pins were used for nrf52840dk and nrf21540 uart0 (console uart).
Due to that debug configuration was not printing anything on that DK.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
drivers: serial: lpuart: Improve logging

Improve debug messages.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
drivers: serial: lpuart: Improve handling of fault cases

Improve handling of a case when control pins do not follow the
protocol behavior (e.g. they are floating). In that case driver
should not trigger assert but just return to the off state.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
samples: peripheral: lpuart: Update README with pin setup

Add details about pin setup for all supported boards.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

alexsven committed:
samples: nrf_auraconfig: Add doc link to lc3-files

- Add entry in README on how to find pre-encoded lc3-files
- Update filepath for usecases
- OCT-3166

Signed-off-by: Alexander Svensen <alexander.svensen@nordicsemi.no>

doki-nordic committed:
samples: esb: Clock control for nRF54h20

This adds support for clock control driver
for nRF54h20 target in the ESB samples.

Jira: NCSDK-28468

Signed-off-by: Dominik Kilian <Dominik.Kilian@nordicsemi.no>
Co-authored-by: Kamil Gawor <Kamil.Gawor@nordicsemi.no>

doki-nordic committed:
samples: dtm: Add clock control for nRF54h20

This adds clock control driver for nRF54h20
target in Direct Test Mode sample.

Jira: NCSDK-28468

Signed-off-by: Dominik Kilian <Dominik.Kilian@nordicsemi.no>
Co-authored-by: Kamil Gawor <Kamil.Gawor@nordicsemi.no>

doki-nordic committed:
samples: radio_test: Add clock control for nRF54h20

This adds support for clock control driver for
nRF54h20 targets in Radio Test sample.

Jira: NCSDK-28468

Signed-off-by: Dominik Kilian <Dominik.Kilian@nordicsemi.no>
Co-authored-by: Kamil Gawor <Kamil.Gawor@nordicsemi.no>

MarkusLassila committed:
applications: serial_lte_modem: Fix timeouts with nativetls

Default timeout for TCP, TLS and DTS connections is 3 seconds,
which does not account for network issues. Increase timeout
to 30 seconds.

Signed-off-by: Markus Lassila <markus.lassila@nordicsemi.no>

ktaborowski committed:
manifest: update sidewalk revision

[KRKNWK-19489]

Signed-off-by: Krzysztof Taborowski <krzysztof.taborowski@nordicsemi.no>

nordic-piks committed:
samples: peripheral: lpuart: verify sample output

Check both ASYNC and IRQ mode.

Signed-off-by: Piotr Kosycarz <piotr.kosycarz@nordicsemi.no>

nordic-piks committed:
test-spec: low-level add lpuart

Extend gpio.

Signed-off-by: Piotr Kosycarz <piotr.kosycarz@nordicsemi.no>

wiba-nordic committed:
doc: zigbee: add deprecation note to all Zigbee-related pages

Zigbee R22 support is deprecated with nRF Connect SDK v2.8.0.
This adds a note to the start of every Zigbee-related page to inform
the user of this.

Signed-off-by: Wille Backman <wille.backman@nordicsemi.no>

nordic-piks committed:
tests: drivers: lpuart: specify needed fixture

Make common tag.

Signed-off-by: Piotr Kosycarz <piotr.kosycarz@nordicsemi.no>

nordic-pikr committed:
tests: benchmarks: power_consumption: temperature sensor

Add test that verify power management for temp sensor

Signed-off-by: Piotr Krzyzanowski <piotr.krzyzanowski@nordicsemi.no>

nordic-pikr committed:
tests: benchmarks: remove GPD workaround

Remove GDP workaround.

Signed-off-by: Piotr Krzyzanowski <piotr.krzyzanowski@nordicsemi.no>

nordic-krch committed:
tests: drivers: lpuart: Fix memory corruption

Data allocated on stack was used as a user data passed to the
interrupt. By the time interrupt is handled this data could be
overwritten leading to a fault. Adding static keyword to that
variable.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

maciejbaczmanski committed:
doc: matter: add missing TFM storage to memory requirements

TFM storage was missing in memory measurements

Signed-off-by: Maciej Baczmanski <maciej.baczmanski@nordicsemi.no>

gmarull committed:
scripts: west_commands: genboard: support current NCS only

Maintaining multiversion compatibility is a nightmare, so lets make the
tool support the current NCS version, ie, the one in the same tree.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

FrancescoSer committed:
doc: Updated 54H GS guide.

Removed mentions of adac discovery cmd.
Updated the guide.

Signed-off-by: Francesco Domenico Servidio <francesco.servidio@nordicsemi.no>

NordicBuilder committed:
manifest: Update sdk-nrfxlib revision (auto-manifest PR)

Automatically created by Github Action

Signed-off-by: Nordic Builder <pylon@nordicsemi.no>

greg-fer committed:
doc: deprecation notes for nrfjprog and nRFCLT

Added deprecation notes for nRF Command Line Tools and nrfjprog.
NCSDK-30028.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

laxiLang committed:
manifest: Update sdk-zephyr revision (auto-manifest PR)

Automatically created by Github Action

Signed-off-by: Lang Xie <lang.xie@nordicsemi.no>

annwoj committed:
doc: flpr support note

Added a note on experimental support

Signed-off-by: Anna Wojdylo <anna.wojdylo@nordicsemi.no>

sarah-read committed:
manifest: Remove ANT revision information

ANT Wireless support is transitioning to the nRF Connect SDK Add-Ons
model from this point forward. Removing the manifest entry for ANT is
the first step in this process. Future ANT for nRF Connect SDK releases
will target tagged nRF Connect SDK revisions with some lag to allow for
alignment. Please contact Garmin Canada at http://www.thisisant.com for
more information on release timelines.

Signed-off-by: Sarah Read <sarah.read@garmin.com>

maje-emb committed:
samples: edge_impulse: Add support for nRF54H20DK

Adds support for nRF54H20DK in the data_forwarder and wrapper samples.

Ref: NCSDK-30037

Signed-off-by: Marcin Jelinski <marcin.jelinski@nordicsemi.no>

alexsven committed:
samples: nrf_auraconfig: Update link to lc3 data

- Created a folder in Artifactory so that the link will never change
- OCT-3166

Signed-off-by: Alexander Svensen <alexander.svensen@nordicsemi.no>

nordic-bami committed:
tests: benchmarks: Correct UARTE test with automatic PM

This test was implemented incorrectly - fix it

Signed-off-by: Bartosz Miller <bartosz.miller@nordicsemi.no>

gmarull committed:
ci: stale: be more agressive

- 30 days instead of 60
- Give 1 week instead of 2
- No label exemption, also delete unused issue labels (NCS does not have
  issues)

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

NordicBuilder committed:
manifest: Update sdk-zephyr revision (auto-manifest PR)

Automatically created by Github Action

Signed-off-by: Nordic Builder <pylon@nordicsemi.no>

jangalda-nsc committed:
ci: Bump pip-compile-cross-platform

pip-compile-cross-platform==1.4.2+nordic2 fixes issues with
unnecessary updates of python packages when they are
available in PYPI.

Signed-off-by: Jan Gałda <jan.galda@nordicsemi.no>

NordicBuilder committed:
manifest: Update sdk-zephyr revision (auto-manifest PR)

Automatically created by Github Action

Signed-off-by: Nordic Builder <pylon@nordicsemi.no>

jangalda-nsc committed:
ci: Change order of input requirements files

Order of input files matter in pip-compile-cross-platform.
The latter file has higher priority and overrides version of
package set in former file.

Signed-off-by: Jan Gałda <jan.galda@nordicsemi.no>

Damian-Nordic committed:
doc: update nRF RPC UART documentation

Improve the text formatting and add "Configuration" and
"Reliability" sections. Also, remove the information that
nRF RPC IPC is used in BLE RPC library, as the latter can
now use both IPC and UART transports.

Signed-off-by: Damian Krolik <damian.krolik@nordicsemi.no>

krish2718 committed:
samples: wifi: Fix 54 series docs

Due to missing integration platforms, 54 Series doesn't show in the
auto generated table in the README.

It was broken in 8ae6472d41 ("samples: wifi: shell: Update
nrf7002eb_interposer tests").

Signed-off-by: Chaitanya Tata <Chaitanya.Tata@nordicsemi.no>

nordic-piks committed:
scripts: quarantine_zephyr: extend for libraries.libc.newlib.thread_safety

region FLASH overflowed

Signed-off-by: Piotr Kosycarz <piotr.kosycarz@nordicsemi.no>

Damian-Nordic committed:
doc: document OpenThread RPC

Add a documentation page about OpenThread RPC.

Signed-off-by: Damian Krolik <damian.krolik@nordicsemi.no>

annwoj committed:
doc: small FOTA fixes

items as per UX session

Signed-off-by: Anna Wojdylo <anna.wojdylo@nordicsemi.no>
